api/errors.ml

```diff
@@ -365,6 +365,20 @@ let fail_lexer_error ~red:_ = function
 let fail_parser_error ~red:_ = function
   | Parser.Parse_error (lc, msg) ->
       Some (702, Some lc, Format.asprintf "Parsing error: %s@." msg)
+  | Preterm.BetaRedexInLHS lc ->
+      Some
+        ( 705,
+          Some lc,
+          Format.asprintf
+            "Parsing error: Beta-redex are not allowed in the left-hand side \
+             of rewriting rule@." )
```

Comment:
On the question of error messages, it would be nice to hint the user how to fix the problem, or put a link to a documentation (provided that we get a stable URL). That said, I don't mind merging this as it is and updating afterwise.

Reply:
Do you think that it is not explicit enough as is?
Regarding your request, it seems to contain 2 very different aspects. 
 - On the hinting part, I could print "why don't you try to replace `((x => f) t)` by `f[y/x]`" (It requires to stay concentrated when implementing the substitution, but it is definitely doable).
 - On the documentation part, there is in any documentation of lambda-pi calculus modulo theory, a grammar of "patterns" which excludes beta-redexes. There is probably even one document which explains why (the reason is probably a confluence criterion, but it might as well be that it is harder/impossible to define Miller's pattern without excluding it). But what are considered as the "official documentations"? And this would explain why this is an issue, but not hints how to solve it.

Reply:
> it is definitely doable
Well, I wrote it a bit fast, implementing substitution is doable, but it could happen that after substitution, the result is still a beta-redex, and hinting to replace one beta-redex by an other might be confusing. But termination of beta-reduction is undecidable, so it might happen that the hinting mechanism turns Dedukti to run forever on a syntactically ill-formed file, probably not a behaviour we want (but really unlikely to be triggered by accident).

Reply:
The message is explicit enough in saying where the problem is, but IMO it is a good practice to hint how to solve it. Your first point makes a lot of sense to me. For the second, I was thinking of the manapage or the readme, nothing more, and detailing the section the user should refer to (with the risk of forgetting to update the reference when the documentation is updated). You're also right that it doesn't hint anything, but it may provide more background information.

In conclusion, your first proposition looks good to me. Doing nothing is also fine.

Reply:
**Answer to second message:** I guess either you're willing to implement something with some fuel, or we leave this for another time.